server/src/main/java/org/elasticsearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -277,10 +351,18 @@ private MergeResult partialReduce(
                 }
             }
             // we have to merge here in the same way we collect on a shard
-            newTopDocs = topDocsList == null ? null : mergeTopDocs(topDocsList, topNSize, 0);
+            newTopDocs = topDocsList == null ? Lucene.EMPTY_TOP_DOCS : mergeTopDocs(topDocsList, topNSize, 0);
             newAggs = hasAggs
-                ? aggregate(toConsume.iterator(), lastMerge, resultSetSize, aggReduceContextBuilder.forPartialReduction())
+                ? aggregate(
+                    toConsume.iterator(),
+                    lastMerge == null ? Collections.emptyIterator() : Iterators.single(lastMerge.reducedAggs),
+                    resultSetSize,
+                    aggReduceContextBuilder.forPartialReduction()
+                )
                 : null;
+            for (QuerySearchResult querySearchResult : toConsume) {
+                querySearchResult.markAsPartiallyReduced();
```

Comment:
does this do anything when we set it in the coord node codepath?

Reply:
Hmm good question :) not right now actually. I'd rather not mess with this right now to keep the noise down. But :) now that you made me think about it, I think it would make sense to either run batched execution locally for combined coord+data nodes (that has some downsides that need benchmarking first) or at least enabled the eager release of contexts (that otoh seems like a freebie :)) via this logic for them as well down the line.

Reply:
My question was only for my clarity , to make sure I am not missing anything, given there's code that is executed on both data nodes and coord nodes, yet in a different context and with a different outcome. No actions to take, thanks for clarifying.